### PR TITLE
392 test reassign anchor properties

### LINKED ISSUE #392: Add tests for `cluster.reassign_gdf_near_anchor_properties`

### PR DESCRIPTION
**Pull request by crispy-wonton**
Fixes #392 

# Description

Test technology reassignment works as expected within, outside, and intersecting with anchor property radius.

Modified files:
- `asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py` - add tests for `cluster.reassign_gdf_near_anchor_properties`
- `asf_heat_pump_suitability/pipeline/cluster/cluster.py` - correct a misleading annotation

# Instructions for Reviewer

In order to test the code in this PR you need to ...
Run the following from your terminal:
```
pytest
```

Please pay special attention to ...
- Any edge cases missing?

# Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

### CODE REVIEW

File: asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py [hunk 1]
```diff
@@ -779,3 +777,60 @@ def test_cells_entirely_contain_buildings_after_overlay(
...
+        )
+
+    def test_cells_within_anchor_radius(self, tech_gdf, gdf_anchor_property):
+        """Test technology reassignment for networked heat pump buildings within anchor property radius."""
+        reassigned_gdf = reassign_gdf_near_anchor_properties(
+            tech_gdf=tech_gdf, combined_anchor_gdf=gdf_anchor_property, radius=1000
+        )
+        results = reassigned_gdf.set_index("building_id").to_dict()["assigned_tech"]
+        expected = tech_gdf.set_index("building_id").to_dict()["assigned_tech"]
+        reassigned_buildings = ["B02", "B03", "B04"]
```
Comment (sofiapinto) [inline asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py:832]: We're checking that only networked solutions are re-assigned. so maybe add a note above this line? to make it more explicit

File: asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py [hunk 2]
```diff
@@ -779,3 +777,60 @@ def test_cells_entirely_contain_buildings_after_overlay(
...
+        """Create single anchor property, 50m east of easternmost networked heat pump building in `gdf_mixed_buildings`."""
+        anchor_property = Polygon(
+            [(400135, 399995), (400145, 399995), (400145, 400005), (400135, 400005)]
+        )
+
+        return gpd.GeoDataFrame(
+            {"class": ["school"], "geometry": [anchor_property]}, crs="EPSG:27700"
+        )
+
+    def test_cells_within_anchor_radius(self, tech_gdf, gdf_anchor_property):
```
Comment (sofiapinto) [inline asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py:824]: note to self to check if we've covering everything: In this case all network solutions are within radius for reassignment

File: asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py [hunk 3]
```diff
@@ -779,3 +777,60 @@ def test_cells_entirely_contain_buildings_after_overlay(
...
+        expected = tech_gdf.set_index("building_id").to_dict()["assigned_tech"]
+        reassigned_buildings = ["B02", "B03", "B04"]
+        for b in reassigned_buildings:
+            expected[b] = "Communal solution"
+
+        assert (
+            results == expected
+        ), "Technology reassignment for building within anchor property radius failed"
+
+    def test_cells_outside_anchor_radius(self, tech_gdf, gdf_anchor_property):
```
Comment (sofiapinto) [inline asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py:840]: note to self to check if we've covering everything: In this case none of the network solutions are within radius

File: asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py [hunk 4]
```diff
@@ -779,3 +777,60 @@ def test_cells_entirely_contain_buildings_after_overlay(
...
+            tech_gdf=tech_gdf, combined_anchor_gdf=gdf_anchor_property, radius=40
+        )
+        results = reassigned_gdf.set_index("building_id").to_dict()["assigned_tech"]
+        expected = tech_gdf.set_index("building_id").to_dict()["assigned_tech"]
+
+        assert (
+            results == expected
+        ), "Unexpected technology reassignment for building outside anchor property radius"
+
+    def test_cells_intersecting_anchor_radius(self, tech_gdf, gdf_anchor_property):
```
Comment (sofiapinto) [inline asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py:852]: note to self to check if we've covering everything: In this case one of the network solution is within the radius, just at the threhsold

File: asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py (hunk 2 above)
Comment (sofiapinto) [reply asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py:824]: might be worth testing an example where only a few are reassigned.

**Review by sofiapinto — COMMENTED** [review]
Great work again @crispy-wonton - see a few comments below. And, as per the previous PR, can we please bring the latest dev changes into the branches so i can run the tests? thank you!

File: asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py (hunk 2 above)
Comment (crispy-wonton) [reply asf_heat_pump_suitability/pipeline/cluster/tests/test_cluster.py:824]: I think together the tests cover all the cases. But lmk if you think there needs to be more cases.

**Review by sofiapinto — APPROVED** [review]
